src/mesh/index_derivs.cxx

```diff
@@ -1164,18 +1170,22 @@ const Field3D Mesh::indexDDX(const Field3D &f, CELL_LOC outloc, DIFF_METHOD meth
       diffloc = (inloc == CELL_CENTRE) ? CELL_XLOW : CELL_CENTRE;
 
     } else {
-      // A more complicated shift. Get a result at cell centre, then shift.
-      if (inloc == CELL_XLOW) {
-        // Shifting
-
-        func = sfDDX;                // Set default
-        table = FirstStagDerivTable; // Set table for others
-        diffloc = CELL_CENTRE;
-
-      } else if (inloc != CELL_CENTRE) {
-        // Interpolate then (centre -> centre) then interpolate
-        return DDX(interp_to(f, CELL_CENTRE), outloc, method);
-      }
+      // Derivative of interpolated field or interpolation of derivative field
+      // cannot be taken without communicating and applying boundary
+      // conditions, so throw an exception instead
+      throw BoutException("Unsupported combination of inloc/outloc in Mesh:indexDDX(Field3D).");
```

Comment:
I've forgotten most of the discussion about this, forgive me, but is it the case that previously, we needed to communicate/apply BCs and weren't doing so? If so, this is fine, but maybe put the inloc/outloc in the exception?

Reply:
> is it the case that previously, we needed to communicate/apply BCs and weren't doing so?
That's right.

I'm adding inloc/outloc to the exception at the moment... Is there a better way to add an enum than as an int? I can't quickly find a nice way of printing the enum value :disappointed:

Reply:
Ah, I'm afraid not. The easiest way is to make a `constexpr std::map<ENUM, std::string>` and look up the name in that, but that might be overkill here.

Reply:
There is a function for that purpose:
https://github.com/boutproject/BOUT-dev/blob/master/include/interpolation.hxx#L41